packages/cli/src/lib/inspect/output.ts

```diff
@@ -1095,3 +1114,129 @@ export const showHook = async (
     throw error;
   }
 };
+
+export const listWaits = async (world: World, opts: InspectCLIOptions = {}) => {
+  if (!opts.runId) {
+    logger.error(
+      'run-id is required for listing waits. Usage: `workflow inspect waits --runId=<id>`'
+    );
+    process.exit(1);
+  }
+
+  if (opts.stepId) {
+    logger.warn(
+      'Filtering by step-id is not supported for waits, ignoring filter.'
+    );
+  }
+  if (opts.workflowName) {
+    logger.warn(
+      'Filtering by workflow-name is not supported for waits, ignoring filter.'
+    );
+  }
+  if (opts.withData) {
+    logger.warn('`withData` flag is ignored when listing waits');
+  }
+
+  try {
+    // Fetch all events for the run with resolveData=false
+    const events = await world.events.list({
+      runId: opts.runId,
```

Comment:
at some point we can move this into the world spec, so the world just lets you list sleeps/waits and can optimize how it's materialized on the backend